packages/expo-notifications/plugin/src/withNotificationsIOS.ts

```diff
@@ -1,11 +1,67 @@
-import { ConfigPlugin, withEntitlementsPlist } from '@expo/config-plugins';
+import {
+  ConfigPlugin,
+  withEntitlementsPlist,
+  IOSConfig,
+  withXcodeProject,
+} from '@expo/config-plugins';
+import fs from 'fs-extra';
+import path from 'path';
 
-export const withNotificationsIOS: ConfigPlugin<{ mode: 'production' | 'development' }> = (
+import { NotificationsPluginProps } from './withNotifications';
+
+type XcodeProject = any;
+
+export const withNotificationsIOS: ConfigPlugin<NotificationsPluginProps> = (
   config,
-  { mode }
+  { mode = 'development', sounds = [] }
 ) => {
-  return withEntitlementsPlist(config, config => {
+  config = withEntitlementsPlist(config, config => {
     config.modResults['aps-environment'] = mode;
     return config;
   });
+  config = withNotificationSounds(config, { sounds });
+  return config;
+};
+
+export const withNotificationSounds: ConfigPlugin<{ sounds: string[] }> = (config, { sounds }) => {
+  return withXcodeProject(config, config => {
+    setNotificationSounds(sounds, {
+      projectRoot: config.modRequest.projectRoot,
+      project: config.modResults,
+    });
+    return config;
+  });
 };
+
+/**
+ * Save sound files to the Xcode project root and add them to the Xcode project.
+ */
+export function setNotificationSounds(
+  sounds: string[],
+  { projectRoot, project }: { project: XcodeProject; projectRoot: string }
+): XcodeProject {
+  if (!Array.isArray(sounds)) {
+    throw new Error(
+      `Must provide an array of sound files in your app config, found ${typeof sounds}.`
+    );
+  }
+
+  const projectName = IOSConfig.XcodeUtils.getProjectName(projectRoot);
+
+  sounds.map((soundFileRelativePath: string) => {
```

Comment:
```suggestion
  for (const soundFileRelativePath of sounds) {
```
since the results are unused.